xio-core/src/main/java/com/xjeffrose/xio/http/Http1ServerCodec.java

```diff
@@ -134,18 +122,14 @@ private void buildResponse(ChannelHandlerContext ctx, Response response, Channel
                 full.headers().http1Headers(false, false),
                 EmptyHttpHeaders.INSTANCE);
       } else {
-        // TODO(CK): TransferEncoding
         // We don't know the size of the message payload so set TransferEncoding to chunked
+        HttpHeaders headers = response.headers().http1Headers(false, false);
```

Comment:
when `Headers` is an instance of `Http2HeadersWrapper` because of an h1 to h2 transition - transfer-encoding is stripped out by `http1Headers()` call ... so we make the transition up front

Reply:
ahh nice and clean and my lazy hack gets to stay!

Reply:
laziness for the win